components/automate-cli/cmd/chef-automate/status.go

```diff
@@ -234,7 +288,233 @@ func runStatusCmd(cmd *cobra.Command, args []string) error {
+		flags.automate = false
+		flags.chefServer = false
+		nodeStatus(flags, POSTGRESQL, infra, statusCmdResults, remoteExe)
+	} else {
+		statusCmdResults <- statusCmdResult{}
+	}
+
+	if flags.opensearch {
+		flags.automate = false
+		flags.chefServer = false
+		flags.postgresql = false
+		nodeStatus(flags, OPENSEARCH, infra, statusCmdResults, remoteExe)
+	} else {
+		statusCmdResults <- statusCmdResult{}
+	}
+}
+
+func nodeStatus(flags *statusCmdFlags, nodeType string, infra *AutomateHAInfraDetails, statusCmdResults chan statusCmdResult, remoteExe RemoteCmdExecutor) {
+	go func(flags statusCmdFlags, statusCmdResults chan<- statusCmdResult) {
+		writer := cli.NewWriter(os.Stdout, os.Stderr, os.Stdin)
+		remoteExe.SetWriter(writer)
+		nodeMap := constructNodeMapForStatus(&flags, infra)
+		cmdResult, err := remoteExe.ExecuteWithNodeMap(nodeMap)
+		statusCmdResults <- statusCmdResult{
+			cmdResult: cmdResult,
+			writer:    writer,
+			nodeType:  nodeType,
+			err:       err,
+		}
+	}(*flags, statusCmdResults)
+}
+
+func printStatusOutput(cmdResult map[string][]*CmdResult, remoteService string, writer *cli.Writer) {
+	writer.Printf("\n=====================================================%s=======================================================\n", "Status on "+remoteService)
+
+	for _, value := range cmdResult {
+		for _, cmdResult := range value {
+			if cmdResult.Error != nil {
+				printStatusErrorOutput(cmdResult, remoteService, writer)
+			} else {
+				writer.Printf("Output for Host IP %s : \n%s", cmdResult.HostIP, cmdResult.Output+"\n")
+				writer.Success("Command is executed on " + remoteService + " node : " + cmdResult.HostIP + "\n")
+			}
+			writer.BufferWriter().Flush()
+		}
+	}
+}
+
+func printStatusErrorOutput(cmdResult *CmdResult, remoteService string, writer *cli.Writer) {
+	isOutputError := false
+	if strings.Contains(cmdResult.Output, "UnhealthyStatusError") {
+		isOutputError = true
+		writer.Failf("Output for Host IP %s : \n%s", cmdResult.HostIP, cmdResult.Output+"\n")
+		writer.Success("Command is executed on " + remoteService + " node : " + cmdResult.HostIP + "\n")
+	}
+
+	if strings.Contains(cmdResult.Output, "DeploymentServiceUnreachableError") {
+		isOutputError = true
+		writer.Failf(CMD_FAIL_MSG, remoteService, cmdResult.HostIP, cmdResult.Output)
+	}
+
+	if !isOutputError {
+		writer.Failf(CMD_FAIL_MSG, remoteService, cmdResult.HostIP, cmdResult.Error.Error())
+	}
+}
+
+func getValueFromChannel(statusCmdResults chan statusCmdResult, printStatusOutput func(map[string][]*CmdResult, string, *cli.Writer)) error {
+	for i := 0; i < 4; i++ {
+		cmdResult := <-statusCmdResults
+		if cmdResult.err != nil {
+			return cmdResult.err
+		} else {
+			if cmdResult.writer != nil {
+				printStatusOutput(cmdResult.cmdResult, cmdResult.nodeType, cmdResult.writer)
+			}
+		}
+	}
+	return nil
+}
+
+func handleManagedServiceErrorForStatusCmd(flags *statusCmdFlags) error {
+
+	if flags.postgresql && flags.opensearch {
+		return nil
```

Comment:
I think this should also return error instead of nil

Reply:
This i added intentionally when someone using managed services and they trigger status for whole infra then they will get status of Automate and CS only and if someone gives --pg or --os flag intentionally in case of managed services then  they will get error msg

Reply:
Done